fix: The form item margin should appear below extraDom.

hongzzz opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is confirmed.
Your pull requests will be merged after one of the collaborators approve.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

close https://github.com/ant-design/ant-design/issues/50026

### 💡 Background and Solution

Removed the div used to expand the error height, wrapped error and extra with a div, and set the minimum height for that div.


### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Form:  Removed the div used to expand the error height, wrapped error and extra with a div, and set the minimum height for that div.      |
| 🇨🇳 Chinese |     Form: 移除了用于撑开error高度的 div，将 errorDom 和 extraDom 用一个 div 包裹，并为该 div 设置了最小高度。      |

hongzzz committed:
fix: form item margin should below extraDom

hongzzz commented:
> ## 👁 Visual Regression Report for PR #50411 Failed ❌
> > **🎯 Target branch:** master ([aa60e53](https://github.com/ant-design/ant-design/commit/aa60e53d4c9acb3f150fcfa36315155c21e93a82))
> > 📖 [View Full Report ↗︎](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50411/visualRegressionReport/report.html)
> 
> Expected (Branch master)	Actual (Current PR)	Diff
> ![form-disabled-input-debug.compact.png](https://camo.githubusercontent.com/a8eae81125e4f2ee20b1a884b559a7a182891031be02c40c36adda121a3874ad/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d64697361626c65642d696e7075742d64656275672e636f6d706163742e706e673f7265663d3864313037346238) form-disabled-input-debug.compact.png	![form-disabled-input-debug.compact.png](https://camo.githubusercontent.com/bba0d4c950c8ab9741249544c6481f56a6fb3516aa1ed88aedfeae388fe889c3/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d64697361626c65642d696e7075742d64656275672e636f6d706163742e706e673f7265663d3864313037346238) form-disabled-input-debug.compact.png	![](https://camo.githubusercontent.com/2723bc7e3eb1968d5f9af276f0196c86108e8c06c87bce5d00f3816bc0023377/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d64697361626c65642d696e7075742d64656275672e636f6d706163742e706e673f7265663d3864313037346238)
> ![form-disabled-input-debug.compact.png](https://camo.githubusercontent.com/c8953e68322e4416875c499621cc51d306948d6ed6efb143a8901fe75a0b7c0d/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d64697361626c65642d696e7075742d64656275672e636f6d706163742e6373732d7661722e706e673f7265663d3864313037346238) form-disabled-input-debug.compact.png	![form-disabled-input-debug.compact.css-var.png](https://camo.githubusercontent.com/daa9b60bcea1370a38d14d750b5f63e614db76dabb718f4368114a301f79e1dd/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d64697361626c65642d696e7075742d64656275672e636f6d706163742e6373732d7661722e706e673f7265663d3864313037346238) form-disabled-input-debug.compact.css-var.png	![](https://camo.githubusercontent.com/48c0e2bb7d3bff0e3845bcdc4da3f94db075e448cb003bf55c5cd426f4c63f68/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d64697361626c65642d696e7075742d64656275672e636f6d706163742e6373732d7661722e706e673f7265663d3864313037346238)
> ![form-disabled-input-debug.dark.png](https://camo.githubusercontent.com/5dbdd35daea2b71b6391394911ba41a206b2d583438490a4fead4a38f8ad5d99/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d64697361626c65642d696e7075742d64656275672e6461726b2e706e673f7265663d3864313037346238) form-disabled-input-debug.dark.png	![form-disabled-input-debug.dark.png](https://camo.githubusercontent.com/1aacca67a557a9d4c781c2e0869b85f5e87f48e24e237455b678911890aedf45/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d64697361626c65642d696e7075742d64656275672e6461726b2e706e673f7265663d3864313037346238) form-disabled-input-debug.dark.png	![](https://camo.githubusercontent.com/296b00e51d8f7a18e01fe85183aff734109aa24c06aba60618c024249e384329/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d64697361626c65642d696e7075742d64656275672e6461726b2e706e673f7265663d3864313037346238)
> ![form-disabled-input-debug.dark.png](https://camo.githubusercontent.com/6034d5dd1b4d2ed9dcfcf9714ba6458d56ab2628f651bbb0fe4c1b134b9bc0ee/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d64697361626c65642d696e7075742d64656275672e6461726b2e6373732d7661722e706e673f7265663d3864313037346238) form-disabled-input-debug.dark.png	![form-disabled-input-debug.dark.css-var.png](https://camo.githubusercontent.com/cc36378d9193bb407019b4c50cc15ea3a7c2eabdfeeca2077b6913ed29ada28f/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d64697361626c65642d696e7075742d64656275672e6461726b2e6373732d7661722e706e673f7265663d3864313037346238) form-disabled-input-debug.dark.css-var.png	![](https://camo.githubusercontent.com/3f520006f19239ef20b6dd74baa9bc6db2ccc54b0c4a9428a947460b34f21873/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d64697361626c65642d696e7075742d64656275672e6461726b2e6373732d7661722e706e673f7265663d3864313037346238)
> ![form-disabled-input-debug.default.png](https://camo.githubusercontent.com/db113a70233708e8f8a6d2da0c5f8bb2a945c163440bd3ca10a516a705996d47/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d64697361626c65642d696e7075742d64656275672e64656661756c742e706e673f7265663d3864313037346238) form-disabled-input-debug.default.png	![form-disabled-input-debug.default.png](https://camo.githubusercontent.com/0b8f3fa4895a4fccfeb1156d96ff10964f1b30fdce4f7b345d599ee496d5f073/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d64697361626c65642d696e7075742d64656275672e64656661756c742e706e673f7265663d3864313037346238) form-disabled-input-debug.default.png	![](https://camo.githubusercontent.com/130cedf8ad8de020d51daa373ba3fff3abe72826fb2a30d73b60083c6ceaa51f/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d64697361626c65642d696e7075742d64656275672e64656661756c742e706e673f7265663d3864313037346238)
> ![form-disabled-input-debug.default.png](https://camo.githubusercontent.com/b26d898ce4a49d20454b68d6b9dd292445d2a19837dfa78c9c89f0ef9531e608/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d64697361626c65642d696e7075742d64656275672e64656661756c742e6373732d7661722e706e673f7265663d3864313037346238) form-disabled-input-debug.default.png	![form-disabled-input-debug.default.css-var.png](https://camo.githubusercontent.com/af828d7e8b4ce27863ccc7756e1f2612da24a4bf28a92596bcd11dc282210fcc/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d64697361626c65642d696e7075742d64656275672e64656661756c742e6373732d7661722e706e673f7265663d3864313037346238) form-disabled-input-debug.default.css-var.png	![](https://camo.githubusercontent.com/9a57d7c86188506944fa280c0d4fe365294dd42e8239ee3d4273116140609172/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d64697361626c65642d696e7075742d64656275672e64656661756c742e6373732d7661722e706e673f7265663d3864313037346238)
> ![form-validate-static.compact.png](https://camo.githubusercontent.com/3aaba08bb2a1eb2c1ca5f31935942671ec2c5c92279c38f6918027f52f477338/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d76616c69646174652d7374617469632e636f6d706163742e706e673f7265663d3864313037346238) form-validate-static.compact.png	![form-validate-static.compact.png](https://camo.githubusercontent.com/dbd12ad46e9dc3a59651c6af4b74a549158486b8642589a09cfbe883e0e7592b/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d76616c69646174652d7374617469632e636f6d706163742e706e673f7265663d3864313037346238) form-validate-static.compact.png	![](https://camo.githubusercontent.com/d645010c556f38e4d54477a03086561f25c9606b177e6ccf1de14be139adddd9/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d76616c69646174652d7374617469632e636f6d706163742e706e673f7265663d3864313037346238)
> ![form-validate-static.compact.png](https://camo.githubusercontent.com/01d6af2d3466864008f55f94ebd67a615e7795cc830e75db6cd5f0c01bfe6345/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d76616c69646174652d7374617469632e636f6d706163742e6373732d7661722e706e673f7265663d3864313037346238) form-validate-static.compact.png	![form-validate-static.compact.css-var.png](https://camo.githubusercontent.com/555795da660970d8883c68893d40ee7c84cdb1a7d33556fbf4013878c2855fbd/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d76616c69646174652d7374617469632e636f6d706163742e6373732d7661722e706e673f7265663d3864313037346238) form-validate-static.compact.css-var.png	![](https://camo.githubusercontent.com/7dec2b1b9fb5b9d00b56269ebc6f1a4ff12fa0ad78f3c944df9c81be22b38aff/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d76616c69646174652d7374617469632e636f6d706163742e6373732d7661722e706e673f7265663d3864313037346238)
> ![form-validate-static.dark.png](https://camo.githubusercontent.com/6cd4fd4b6204bf321062406c3fc235c6906bba9453027c10b83cfcb2c3783095/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d76616c69646174652d7374617469632e6461726b2e706e673f7265663d3864313037346238) form-validate-static.dark.png	![form-validate-static.dark.png](https://camo.githubusercontent.com/967c78c0b07ed7c52feea4ee4b9587cc13a213d2ad25295bf9cca7c1e4d4dfb3/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d76616c69646174652d7374617469632e6461726b2e706e673f7265663d3864313037346238) form-validate-static.dark.png	![](https://camo.githubusercontent.com/82d63a1532b5135779994c455a6f8406369872f35c91dbeecb8213657e02c58a/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d76616c69646174652d7374617469632e6461726b2e706e673f7265663d3864313037346238)
> ![form-validate-static.dark.png](https://camo.githubusercontent.com/ec61a44c18d9111ad4dca4c80dec060c9c49349509698bbc293e20ce0e5bad31/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f626173652f666f726d2d76616c69646174652d7374617469632e6461726b2e6373732d7661722e706e673f7265663d3864313037346238) form-validate-static.dark.png	![form-validate-static.dark.css-var.png](https://camo.githubusercontent.com/8776fa5a92b7ad1a375f35241147d03eada29d4c58d6c3bbfa13fd41c63747ea/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f63757272656e742f666f726d2d76616c69646174652d7374617469632e6461726b2e6373732d7661722e706e673f7265663d3864313037346238) form-validate-static.dark.css-var.png	![](https://camo.githubusercontent.com/d7f84c38710e69151f1db95a47f94b456065ea09680d1909709324a1fba4c156/68747470733a2f2f616e74642d76697375616c2d646966662e6f73732d636e2d7368616e676861692e616c6979756e63732e636f6d2f70722d35303431312f76697375616c52656772657373696f6e5265706f72742f696d616765732f646966662f666f726d2d76616c69646174652d7374617469632e6461726b2e6373732d7661722e706e673f7265663d3864313037346238)
> Check [Full Report](https://antd-visual-diff.oss-cn-shanghai.aliyuncs.com/pr-50411/visualRegressionReport/report.html) for details
> 
> If you think the visual diff is acceptable, please check:
> 
> * [ ]  Visual diff is acceptable

I'll check it tonight.

yoyo837 commented on components/form/FormItemInput.tsx:
```diff
@@ -109,8 +106,12 @@ const FormItemInput: React.FC<FormItemInputProps & FormItemInputMiscProps> = (pr
-        {errorListDom}
-        {extraDom}
+        <div
+          style={{ display: 'flex', flexDirection: 'column', minHeight: marginBottom || 'unset' }}
```
这个能cssinjs不写行内吗？

li-jia-nan commented:
这是个 breaking change 吧

hongzzz commented:
> 这是个 breaking change 吧

可能算是，因为改动了结构

li-jia-nan commented:
> 可能算是，因为改动了结构； 这里还漏了一种 _internalItemRender 的场景，这两天我再看下怎么解

breaking change 提交到 feature 分支吧

hongzzz committed:
style(form): move style to css

hongzzz committed:
fix: help can be empty string

li-jia-nan commented on components/form/FormItemInput.tsx:
```diff
@@ -71,22 +71,20 @@ const FormItemInput: React.FC<FormItemInputProps & FormItemInputMiscProps> = (pr
-        </FormItemPrefixContext.Provider>
-        {!!marginBottom && <div style={{ width: 0, height: marginBottom }} />}
-      </div>
+    errors.length || warnings.length || hasHelp ? (
```
```suggestion
    errors.length > 0 || warnings.length > 0 || hasHelp ? (
```

hongzzz committed:
chore: Update components/form/FormItemInput.tsx

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: hongzzz <hongzzz@foxmail.com>

afc163 commented on components/auto-complete/__tests__/__snapshots__/demo.test.tsx.snap:
```diff
@@ -383,6 +386,9 @@ exports[`renders components/auto-complete/demo/form-debug.tsx correctly 1`] = `
         </div>
+        <div
+          class="ant-form-item-additional"
+        />
```
为啥会多一个空 div？

afc163 replied:
最好判断一下，避免对 snap 不必要的影响。

hongzzz committed:
chore: additionalDom only show when errorListDom or extraDom exists.

Wxh16144 committed:
fix(SubMenu): popupStyle not working (#50922)

github-actions[bot] committed:
chore: upgrade deps (#50925)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

kiner-tang committed:
docs: Resolve the link redirection error in the Upload (#50929)

thinkasany committed:
demo(form):  improve layout demo (#50933)

yoyo837 commented:
再同步一次源分支吧

github-actions[bot] committed:
chore: upgrade deps (#50940)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

li-jia-nan committed:
Merge branch 'feature' into fix/form-item-margin

dependabot[bot] committed:
chore(deps-dev): bump the dev-dependencies with 1 update (#50938)

* chore(deps-dev): bump the dev-dependencies with 1 update

Bumps the dev-dependencies group with 2 updates: [@antv/g6](https://github.com/antvis/g6) and [@inquirer/prompts](https://github.com/SBoudrias/Inquirer.js).


Updates `@antv/g6` from 4.8.24 to 5.0.23
- [Release notes](https://github.com/antvis/g6/releases)
- [Changelog](https://github.com/antvis/G6/blob/v5/CHANGELOG.md)
- [Commits](https://github.com/antvis/g6/commits)

Updates `@inquirer/prompts` from 5.5.0 to 6.0.1
- [Release notes](https://github.com/SBoudrias/Inquirer.js/releases)
- [Commits](https://github.com/SBoudrias/Inquirer.js/compare/@inquirer/prompts@5.5.0...@inquirer/prompts@6.0.1)

---
updated-dependencies:
- dependency-name: "@antv/g6"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: "@inquirer/prompts"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update package.json

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: lijianan <574980606@qq.com>

kiner-tang committed:
docs: Optimize the operation experience of the demo modal (#50943)

kiner-tang committed:
docs: optimize demo style of lunar (#50944)

github-actions[bot] committed:
chore: auto merge branches (#50945)

chore: merge feature into master

li-jia-nan committed:
chore: fix the Warning of Alert demo (#50946)

thinkasany committed:
fix(select): background color appears when allowClear and filled (#50916)

kiner-tang committed:
fix(Tour): solve the Tour title will cover the close button at a certain length issue (#50942)

github-actions[bot] committed:
chore: auto merge branches (#50953)

chore: merge feature to master

afc163 committed:
docs(🔥): release 5.21.0 with Splitter component and variant buttons (#50948)

Co-authored-by: lijianan <574980606@qq.com>

thinkasany committed:
docs: update changelog (#50959)

zombieJ committed:
docs: blog picker (#50936)

* chore: blog picker

* docs: trigger update

* docs: add ref faq

* docs: update

zombieJ committed:
docs: add desc (#50967)

zombieJ committed:
fix: Table in Transfer (#50974)

afc163 committed:
site: fix suspense fallback when toggle debug mode (#50985)

ytahirkose committed:
fix: tr value added on filterCheckall keyword (#51000)

Co-authored-by: ytahirkose <yasar.kose@elsitel.com.tr>

afc163 committed:
chore: fix emotion style missing in visual diff (#50995)

coding-ice committed:
fix(button): type=link error token & add prefix cls (color/variant) & specificity too high (#50962)

github-actions[bot] committed:
chore: upgrade deps (#51004)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

RomaLetodiani committed:
docs: Update index.en-US.md (#51006)

afc163 committed:
chore: fix duplicated action name (#50994)

afc163 committed:
site: lazy load g6 (#50991)

afc163 committed:
feat: Tree add indentWidth token (#50989)

afc163 committed:
Revert "feat: Tree add indentWidth token" (#51009)

afc163 committed:
docs: release 5.21.1 (#51012)

dependabot[bot] committed:
chore(deps-dev): bump the dev-dependencies group with 1 update (#51021)

* chore(deps-dev): bump the dev-dependencies group with 1 update

Bumps the dev-dependencies group with 2 updates: [@antv/g6](https://github.com/antvis/g6) and [vanilla-jsoneditor](https://github.com/josdejong/svelte-jsoneditor).


Updates `@antv/g6` from 4.8.24 to 5.0.23
- [Release notes](https://github.com/antvis/g6/releases)
- [Changelog](https://github.com/antvis/G6/blob/v5/CHANGELOG.md)
- [Commits](https://github.com/antvis/g6/commits)

Updates `vanilla-jsoneditor` from 0.23.8 to 1.0.0
- [Release notes](https://github.com/josdejong/svelte-jsoneditor/releases)
- [Changelog](https://github.com/josdejong/svelte-jsoneditor/blob/main/CHANGELOG.md)
- [Commits](https://github.com/josdejong/svelte-jsoneditor/compare/v0.23.8...v1.0.0)

---
updated-dependencies:
- dependency-name: "@antv/g6"
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
- dependency-name: vanilla-jsoneditor
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update package.json

Signed-off-by: lijianan <574980606@qq.com>

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: lijianan <574980606@qq.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: lijianan <574980606@qq.com>

thinkasany committed:
site: improve pronunciation (#51017)

yxplus1116 committed:
docs:Remove duplicate definitions of attributes icon and disabled (#51030)

wanpan11 committed:
fix: Splitter.Panel style is invalid (#51032)

coding-ice committed:
fix(select): Incorrect activeBorderColor token when variant is filled (#51054)

github-actions[bot] committed:
chore: upgrade deps (#51056)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

gennadiipel committed:
type: Remove unused className from Tour props (#51059)

sakuraee committed:
fix: add touch listener to Splitter (#51060)

Co-authored-by: root <root@DESKTOP-CIPQ4R3>

thinkasany committed:
site: improve component change docs (#51040)

nathanlao committed:
fix: search component inconsistent height of button and input (#50926)

Co-authored-by: nathanlao11 <nathanl@emonster.ca>
Co-authored-by: afc163 <afc163@gmail.com>

renovate[bot] committed:
chore(deps): update dependency @npmcli/run-script to v9 (#51064)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

afc163 committed:
fix: Typography copyable icon not aligned (#51066)

afc163 committed:
style: tweak lineWidthFocus to 3px (#51069)

afc163 committed:
fix: upgrade rc-tabs (#51072)

afc163 committed:
refactor: remove TransButton in Table/Transfer/Typography (#51068)

thinkasany committed:
site: add version update date for component changedoc (#51070)

afc163 committed:
refactor: refactor components/upload/UploadList/index.tsx (#51074)

chenshuai2144 committed:
docs: update changelog for version 5.21.2 release (#51078)

* dcos: update changelog for version 5.21.2 release

* chore: add empty line before version 5.21.2 in changelog

* chore: revert changes to fix Typography `copyable` icon alignment issue

* chore: update version to 5.21.2 in package.json

* fix: update changelog to correct activeBorderColor token description for Select component

* fix: update changelog to specify Input.Search component for alignment issue

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* fix: reorganize Splitter section in changelog for clarity and fix formatting issues

# Conflicts:
#	CHANGELOG.zh-CN.md

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# interactive rebase in progress; onto 86489ccab0
# Last command done (1 command done):
#    pick 33c6cfef84 fix: reorganize Splitter section in changelog for clarity and fix formatting issues
# No commands remaining.
# You are currently rebasing branch 'changelog' on '86489ccab0'.
#
# Changes to be committed:
#	modified:   CHANGELOG.en-US.md
#	modified:   CHANGELOG.zh-CN.md
#

* Update CHANGELOG.en-US.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* Update CHANGELOG.en-US.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: 陈帅 <qixian.cs@outlook.com>

* fix: 移除关于 Splitter 在触屏设备上拖拽异常的修复记录

---------

Signed-off-by: 陈帅 <qixian.cs@outlook.com>
Co-authored-by: afc163 <afc163@gmail.com>

thinkasany committed:
site: support bold for change doc (#51076)

afc163 committed:
docs: move change-log link to meta part (#51085)

li-jia-nan committed:
chore: update the publish date (#51086)

chentsulin committed:
fix: add peerDependencies back (#51079)

li-jia-nan committed:
site: site style update (#51088)

afc163 committed:
chore: import rc-util from path (#51087)

li-jia-nan committed:
site: audio icon is not aligned of site (#51094)

kiner-tang committed:
fix(Splitter): fix the issue about Splitter had unexpected gaps in Flex (#51096)

akshitbansal2005 committed:
chore: Update webpack.config.js (#51098)

mshobana committed:
docs: clear description of Collapsible property of Collapse (#51102)

Previously the description was not all understandable until I see the examples.

Signed-off-by: mshobana <51772793+mshobana@users.noreply.github.com>

afc163 committed:
docs: add codesandbox sample for findDOMNode is deprecated (#51101)

Zt448143356 committed:
docs: fix notification import statement (#51105)

Co-authored-by: chijing.zt <chijing.zt@alibaba-inc.com>

kiner-tang committed:
fix: fix the issue about throw a warning when Splitter nested in a hidden tab panel (#51109)

kiner-tang committed:
docs: remove deprecated props from document (#51138)

nathanlao committed:
fix: sorter columnkey should be null (#51114)

Cameron-Asdf committed:
style: Make tall dropdowns have scroll bar (#51112)

dependabot[bot] committed:
chore(deps-dev): bump @inquirer/prompts from 6.0.1 to 7.0.0 in the dev-dependencies group (#51141)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

yoyo837 committed:
chore: update slider patch to fix warning (#51150)

* chore: update slider patch

* update snap

thinkasany committed:
site: fix display of refs (#51153)

* site: fix display of refs

* fix

* fix

thinkasany committed:
type(slider): improve eventName type (#51156)

zombieJ committed:
fix: ColorPicker gradient slider not draggable (#51161)

* fix: fill should trigger same color

* test: add test case

* chore: add desc

* chore: force CI

* chore: force CI

MadCcc committed:
docs: changelog 5.21.3 (#51147)

* docs: changelog 5.21.3

* docs: fix changelog format

* chore: update version

* Update CHANGELOG.en-US.md

Signed-off-by: MadCcc <1075746765@qq.com>

* Apply suggestions from code review

Co-authored-by: lijianan <574980606@qq.com>
Signed-off-by: MadCcc <1075746765@qq.com>

* chore: update docs

* docs: update changelog

* docs: update changelog

* docs: update changelog

* Update CHANGELOG.en-US.md

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: MadCcc <1075746765@qq.com>

* Apply suggestions from code review

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: MadCcc <1075746765@qq.com>

* docs: update changelog

---------

Signed-off-by: MadCcc <1075746765@qq.com>
Co-authored-by: lijianan <574980606@qq.com>
Co-authored-by: afc163 <afc163@gmail.com>

lolWK committed:
docs: remove duplicated backticks (#51168)

idealHyun committed:
docs: fixed the typo by changing 'three' to 'third' (#51172)

zombieJ committed:
fix: mix usage should not scroll (#51169)

rlaehd62 committed:
docs: fixed typo by adding 'developers' and 'they' (#51176)

thinkasany committed:
site: add version title href (#51180)

afc163 committed:
docs: tweak demo toggle button style (#51184)

thinkasany committed:
site: show DocAnchor for en changedoc (#51185)

thinkasany committed:
type(switch): improve eventHandler type (#51165)

Meowu committed:
fix(Tree): align switcher icon of tree component. (#51181)

bmmmmmmmmm committed:
docs: the default value of textAlign in watermark-cn (#51212)

Co-authored-by: aojunhao123 <aojunhao123@users.noreply.github.com>

afc163 committed:
ifx: upgrade @ant-design/cssinjs-utils (#51214)

Signed-off-by: afc163 <afc163@gmail.com>

coding-ice committed:
style(button): modify the text suspension bgColor (#51187)

afc163 committed:
docs: use codesandbox as default reproduce link (#51219)

iqingting committed:
fix(input.search): ensure borderColor inherits from theme token (#51226)

zombieJ committed:
docs: 5.21.4 changelog (#51233)

* docs: 5.21.4 changelog

* docs: fix typo

* Update CHANGELOG.zh-CN.md

Co-authored-by: thinkasany <480968828@qq.com>
Signed-off-by: 二货爱吃白萝卜 <smith3816@gmail.com>

* Update CHANGELOG.en-US.md

Co-authored-by: thinkasany <480968828@qq.com>
Signed-off-by: 二货爱吃白萝卜 <smith3816@gmail.com>

---------

Signed-off-by: 二货爱吃白萝卜 <smith3816@gmail.com>
Co-authored-by: thinkasany <480968828@qq.com>

afc163 committed:
docs: improve homepage banner card style (#51221)

wanpan11 committed:
fix: Splitter token rename (#51223)

Co-authored-by: afc163 <afc163@gmail.com>

afc163 committed:
chore: Rename pr-check-ci.yml to pr-auto-merge.yml (#51235)

thinkasany committed:
fix(Typography): Link can be selected by the user (#51243)

thinkasany committed:
docs: improve changedoc (#51244)

Wxh16144 committed:
style(badge): fix text style error (#51252)

hongzzz committed:
Merge branch 'ant-design:master' into fix/form-item-margin

hongzzz commented:
已同步
